src/toil/job.py

```diff
@@ -1282,7 +1326,11 @@ def __init__(self, job):
         :param toil.job.Job job: the job to encapsulate.
         """
         # Giving the root of the subgraph the same resources as the first job in the subgraph.
-        Job.__init__(self, disk=job.disk, memory=job.memory, cores=job.cores)
+        # The service resource requirments are not guaranteed to be set yet so default to None
+        Job.__init__(self, memory=getattr(job, 'memory', None),
```

Comment:
This makes me curious. Under what circumstance can `job` not have the `memory` attribute?